src/compiler/checker.ts

```diff
@@ -15459,6 +15463,39 @@ namespace ts {
                     }
                 }
             }
+
+            function getUnImplementedAncestorMembers(type: InterfaceType): SymbolTable {
+                const results: SymbolTable = {};
+
+                const baseTypes = getAbstractAncestorTypes(type);
+                for (const bt of baseTypes) {
+                    const abstractMembers = getNamedMembers(bt.symbol.members).filter(isSymbolAbstract);
+                    for (const am of abstractMembers) {
+                        results[am.name] = am;
+                    }
+                }
+
+                for (const member of getPropertiesOfType(type).filter(m => !isSymbolAbstract(m))) {

```

Comment:
Avoid the double iteration - this is worse because you're also mutating your results.